aten/src/ATen/native/TensorFactories.h

```diff
@@ -65,8 +65,10 @@ inline void check_args(
 
 using at::check_size_nonnegative;
 
+// assumes maximum value in created tensor is n-1 (e.g., torch.randperm(n))
 inline void check_supported_max_int_with_precision(int64_t n, const Tensor& tensor) {
-  TORCH_CHECK(at::scalar_tensor(n, tensor.options()).defined(),
+  // match defined() to behavior of checks below
+  TORCH_CHECK(at::scalar_tensor(n - 1, tensor.options()).defined(),
```

Comment:
what would happen if n=0 here?

Reply:
good catch, I tried to avoid having to write `n>0?`...